### DISCUSSION TITLE
Cancel WhenAny - linked CancellationTokenSource

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#106] [General]
# Cancel WhenAny - linked CancellationTokenSource

In today's short post, I will show you how to cancel a Task with multiple inputs (aka multiple CancellationTokens) using CancellationTokenSource and CancellationTokenSource.CreateLinkedTokenSource.

https://steven-giesel.com/blogPost/cd950c5f-467b-48a3-99fe-e72865460579

**Comment by jslicer** [comment]
So I'm looking through one of my company's codebases, and I see a number of these which only seem to link one token, like this:

```csharp
CancellationTokenSource cts = CancellationTokenSource.CreateLinkedTokenSource(this._ct);
```

This seems to allow a more "local" method (couldn't find a better term, not talking about local methods) to call `cts.Cancel()` and/or the single linked token to be cancelled from elsewhere. Is my understanding correct?

**Reply by linkdotnet** [reply]
Without more context, I can only speculate. But this approach lets you listen to two Tokens: `this._ct` and `cts.Token`. If one of them is getting canceled, your task gets canceled. This also allows active cancellation of `cts` without interfering with `this._ct,` which might cancel other tasks.
